src/datanode/src/datanode.rs

```diff
@@ -246,7 +246,8 @@ impl DatanodeBuilder {
 
         info!("going to open {} regions", regions.len());
 
-        for (region_id, engine, region_dir) in regions {
+        for (region_id, engine, path) in regions {
```

Comment:
Great catch! It's better to add a test to cover this case.

Reply:
Let's leave this as homework to @waynexia 😚